dlib/cuda/cudnn_dlibapi.cpp

```diff
@@ -162,6 +162,12 @@ namespace dlib
             return des.get_handle();
         }
 
+        static cudnnActivationDescriptor_t mish_activation_descriptor()
+        {
+            thread_local cudnn_activation_descriptor des(CUDNN_ACTIVATION_MISH, CUDNN_PROPAGATE_NAN,0);
```

Comment:
This needs to have #ifdef around it like I noted earlier.  For instance, I just tried to build this with cudnn 10.1 and it failed to build because this feature is not available.

Reply:
Really though, it's easier to just delete the cudnn calls and use your own cuda code.  You already have the mish function, you just need the gradient and you are good to go.

Reply:
Where is `CUDNN_ACTIVATION_MISH` even available? Downloaded latest cuDNN, but couldn't find anything.

Reply:
Yeah it's not available. I need to change this and use the cuda code only.

Reply:
Fixed by adding a standalone CUDA implementation.